extensions/wikia/ArticleComments/classes/ArticleComment.class.php

```diff
@@ -1588,4 +1522,93 @@ static private function isBadAccessError( $errors ) {
 		return in_array( 'badaccess-groups', $errors ) || in_array( 'badaccess-group0', $errors );
 	}
 
+	/**
+	 * Manages permissions related to article and blog comments
+	 * Hook: userCan
+	 *
+	 * @param Title $title
+	 * @param User $user
+	 * @param string $action
+	 * @param bool $result Whether $user can perform $action on $title
+	 * @return bool Whether to continue checking hooks
+	 */
+	static public function userCan( Title &$title, User &$user, $action, &$result ) {
+		$app = F::app();
+		$commentsNS = $app->wg->ArticleCommentsNamespaces;
+		$ns = $title->getNamespace();
+
+		// Only handle article and blog comments
+		if ( !in_array( MWNamespace::getSubject( $ns ), $commentsNS ) ||
+			!ArticleComment::isTitleComment( $title ) ) {
+			return true;
+		}
+
+		$comment = ArticleComment::newFromTitle( $title );
+		$isBlog = ( $app->wg->EnableBlogArticles && ArticleComment::isBlog( $title ) );
+
+		$return = true;
+		switch ( $action ) {
+			// Creating article comments requires 'commentcreate' permission
+			// For blogs, additionally check if the owner has enabled commenting+
+			case 'create':
+				// We have to check these permissions on the parent article
+				// due to the chicken-and-egg problem inherent in the design
+				$result = self::userCanCommentOn( $comment->getArticleTitle(), $user );
+				$return = false;
+				break;
+			// Article and blog comments can only be edited by their author,
+			// or an user with 'commentedit' permission
+			case 'edit':
+				// Prepopulate the object with revision data
+				// required by ArticleComment::isAuthor
+				$comment->load( true );
+				$result = ( $comment->isAuthor( $user ) || $user->isAllowed( 'commentedit' ) );
+				$return = false;
+				break;
+
+			case 'move':
+			case 'move-target':
+				$result = $user->isAllowed( 'commentmove' );
+				$return = false;
+				break;
+
+			case 'delete':
+				$result = ( ArticleComment::isTitleComment( $title ) &&
+					( $user->isAllowed( 'commentdelete' ) || $isBlog && $user->isAllowed( 'blog-comments-delete' ) )
+				);
+				$return = false;
+			default:
+		}
+
+		return $return;
+	}
+
+	/**
+	 * Check if user can add a comment to the current article
+	 * We must perform this check on the article
+	 * because of the chicken-and-egg problem inherent in the design
+	 *
+	 * @param Title $title Article title
+	 * @param User|null $user Current user
+	 * @return bool Whether $user can add a comment to $title
+	 */
+	static public function userCanCommentOn( Title $title, User $user = null ) {
```

Comment:
I don't think we need to do this here, `Title` internally passes `$this` to the handlers, so we should only check that we don't call `->userCan()` on a non-object.